### ISSUE TITLE
not able to show the code documentation of asInstanceOf and other synthetic methods

### ISSUE THREAD

**Issue by robstoll** [scalameta/metals#6153]
### Describe the bug

I would like to be able to see the documentation of asInstanceOf.
Most likely it is not shown since Any.class is a synthetic class and not present in the scala-library.jar neither in scala-library-sources.jar. 
See also https://youtrack.jetbrains.com/issue/SCL-22167/documentation-missing-for-methods-defined-in-Any

### Expected behavior

I can see the documentation as for any other method

### Operating system

Linux

### Editor/Extension

VS Code

### Version of Metals

v1.29.0

**Comment by tgodzik** [comment]
Thanks for reporting! Looks like the sources might not contain them. We could try getting documentation from https://github.com/scala/scala/blob/fcc67cd56c67851bf31019ec25ccb09d08b9561b/src/library-aux/scala/Any.scala#L150 somehow

**Comment by robstoll** [comment]
Yes, the sources.jar does not contain it. Jetbrains suggested to add it to the sources.jar. Would that be enough for metals to pick it  up?

**Comment by tgodzik** [comment]
I think so. We do index everything there. We could possibly download it additionally ourselves if needed and is published

**Comment by bishabosha** [comment]
in Scala 3 at least they should be available in the `ctx.docCtx` in the compiler

**Comment by robstoll** [comment]
will be in sources.jar with https://github.com/scala/bug/issues/12958

**Comment by kasiaMarek** [comment]
Fixed upstream and it works now. 

<img width="871" alt="Image" src="https://github.com/user-attachments/assets/60b277dc-f45f-46d2-8507-a8b63fa78187" />
